### ISSUE TITLE
remix-auth-oauth2 broken on main with current version of @oslojs/oauth2

### ISSUE THREAD

**Issue by runjak** [sergiodxa/remix-auth-oauth2#106]
It seems that the current (`0fcee48`) state of main has [imports from @oslojs/oauth2](https://github.com/sergiodxa/remix-auth-oauth2/blob/0fcee480340c5d0e8d4bab7dcaefc0b8ad351a51/src/index.ts) that don't match the [3 currently exported classes](https://github.com/oslo-project/oauth2/blob/40575e3f9e10c42c07c92540ac515d1821e62b09/src/index.ts) in [version ^0.5.0](https://github.com/sergiodxa/remix-auth-oauth2/blob/0fcee480340c5d0e8d4bab7dcaefc0b8ad351a51/package.json#L51).

It is my understanding that as a result this projects main branch has been broken since about 2024-07-15.

When building I get the following errors:

```bash
src/index.ts(2,2): error TS2305: Module '"@oslojs/oauth2"' has no exported member 'AuthorizationCodeAccessTokenRequestContext'.
src/index.ts(3,2): error TS2305: Module '"@oslojs/oauth2"' has no exported member 'AuthorizationCodeAuthorizationURL'.
src/index.ts(4,2): error TS2305: Module '"@oslojs/oauth2"' has no exported member 'OAuth2RequestError'.
src/index.ts(5,2): error TS2305: Module '"@oslojs/oauth2"' has no exported member 'RefreshRequestContext'.
src/index.ts(6,2): error TS2305: Module '"@oslojs/oauth2"' has no exported member 'TokenResponseBody'.
src/index.ts(7,2): error TS2305: Module '"@oslojs/oauth2"' has no exported member 'TokenRevocationRequestContext'.
src/index.ts(8,2): error TS2305: Module '"@oslojs/oauth2"' has no exported member 'generateCodeVerifier'.
src/index.ts(9,2): error TS2305: Module '"@oslojs/oauth2"' has no exported member 'generateState'.
src/index.ts(10,2): error TS2305: Module '"@oslojs/oauth2"' has no exported member 'sendTokenRequest'.
src/index.ts(11,2): error TS2305: Module '"@oslojs/oauth2"' has no exported member 'sendTokenRevocationRequest'.
src/index.ts(449,15): error TS2305: Module '"@oslojs/oauth2"' has no exported member 'TokenResponseBody'.
```

It also seems that `@oslo/oauth2` in 0.5.0 [dropped its dependency](https://github.com/oslo-project/oauth2/commit/3eb7ef31f1d6580277d96f50ea26cabd1966687d#diff-7ae45ad102eab3b6d7e7896acd08c427a9b25b346470d7bc6507b6481575d519) on [@oslojs/crypto](https://crypto.oslojs.dev/) and did so far not replace it with anything instead. This makes me wonder what will take the place of `generateCodeVerifier` and similar code in the future.

Short term it might be helpful to roll back the version change so that main can be built again.
Longer term it seems that there's an implicit question here how `@oslo/oauth2` will develop and so on.

**Comment by sergiodxa** [comment]
Fixed in #108

**Comment by runjak** [comment]
Ah thank you 🙏
